GaiaTileSet.js

```diff
@@ -1,57 +1,104 @@
-const fs = require('fs');
 const path = require('path');
 const LRU = require('lru-cache');
-const _latLng = require('./node_modules/node-hgt/src/latlng');
-const tileKey = require('./node_modules/node-hgt/src/tile-key');
-const Hgt = require('./node_modules/node-hgt/src/hgt');
+const {format} = require('util')
+const {coordAll, coordEach} = require('@turf/meta');
+const HGT = require('./hgt');
+const getHGTElevation = require('./hgt/getHGTElevation');
 
-function TileSet(tileDir, NO_DATA = 0) {
+function GaiaTileSet(tileDir, NO_DATA = 0) {
     this._tileDir = tileDir;
     this._cache = new LRU({
         // 500mb
         max: 500000000,
         // 30 days
         maxAge: 1000 * 60 * 60 * 24 * 30,
-        length: n => n._buffer.length,
+        length: n => n.buffer.length,
         updateAgeOnGet: true,
+        dispose: (key, value) => {
+            // pass
+        }
     });
     this._NO_DATA = NO_DATA;
 }
 
-TileSet.prototype.destroy = function() {};
+GaiaTileSet.prototype.destroy = function() {
+    this._cache.reset();
+};
 
-TileSet.prototype._loadTile = function(tileDir, latLng) {
-    const ll = {
-        lat: Math.floor(latLng.lat),
-        lng: Math.floor(latLng.lng)
-    }
-    const key = tileKey(ll);
-    const tilePath = path.join(tileDir, key + '.hgt');
+// Get the appropriate HGT tile for a given coordinate
+GaiaTileSet.prototype._loadTile = function(coord, callback) {
+    coord = coord.map(Math.floor);
 
+    const key = getTileKey(coord);
     const cachedTile = this._cache.get(key);
-    if (cachedTile) return [undefined, cachedTile];
-
-    const tileExists = fs.existsSync(tilePath);
-    if (!tileExists) return [{message: 'Tile does not exist: ' + tilePath}]
+    if (cachedTile) return callback(undefined, cachedTile);
 
+    const tilePath = path.join(this._tileDir, key + '.hgt');
     try {
-        const tile = new Hgt(tilePath, ll);
-        this._cache.set(key, tile);
-        return [undefined, tile];
+        HGT(tilePath, coord, undefined, (error, tile) => {
+            setImmediate(() => {
+                if (error) return callback([{message: 'Tile does not exist'}])
+                this._cache.set(key, tile);
+                callback(undefined, tile);
+            })
+        })
     } catch(e) {
-        return [{message: 'Unable to load tile "' + tilePath + '": ' + e, stack: e.stack}];
+        callback({message: 'Unable to load tile "' + tilePath + '": ' + e, stack: e.stack});
```

Comment:
under what conditions does HGT() throw now (and not call callback)?

Reply:
It will only `throw` when an unsupported data file is encountered. However, that seems more analogous to a data file not being found, which returns an error. I'll update `getResolutionAndSize` to return an error instead of throwing.